lib/extensions/pathname_helpers/pathname.rb

```diff
@@ -0,0 +1,28 @@
+module Extensions::PathnameHelpers::Pathname
+  module ClassMethods
+    # Normalizes the filename.
+    #   Replace following characters with whitespace: / \ ? * : | " < >
+    #   Remove leading and trailing whitespaces.
+    #   Collapse intra-string whitespaces into single whitespace.
+    #   Remove trailing dots.
+    #
+    # @param [String] filename
+    # @return [String] The normalized filename.
+    def normalize_filename(filename)
+      filename.tr('\\\/\"\?\*\:\|<>', ' ').strip.gsub(/\s+/, ' ').gsub(/\.+$/, '')
```

Comment:
You should check what single-quoted strings mean, I don't think they need to be doubly escaped.